### PR TITLE
Fix reading DICOM files with extra tags in the file meta information group

### LINKED ISSUE #1376: When a DICOM file contains two meta information segments, the second segment overwrites the first
**Describe the bug**
When a DICOM file is opened  that has multiple instances of group 0x0002 (= File Meta Information),
the file meta information is taken from the last instance instead of the first.
I'm aware that this file is technically invalid, but we see instances of this in the wild and other tools like dcmdump can properly handle them.
We noticed this because, when trying to render such a DICOM file, the wrong transfer syntax (from the 2nd segment) is used by fo-dicom.

**To Reproduce**
I will add a unit test that demonstrates the problem, once I figure out how to anonymize the sample file properly.

**Expected behavior**
The second (and any other) instances of that group should be considered tags belonging to the data, not the meta information.

**Environment**
Fellow Oak DICOM version: Latest 5.x
OS: Any
Platform: Any

### PR DESCRIPTION
**Pull request by amoerie**
Fixes #1376

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Stop reading file meta information when encountering an tag of group 2 with a lower element value than the previous tag
- To implement this, I also have to keep track of the previous DICOM tag in DicomReader. Then, I updated the FileMetaInformationStop method to also stop when the next tag is suddenly a lower element than the previous tag. This fixes reading the faulty file.

### CODE REVIEW

File: FO-DICOM.Core/IO/Reader/DicomReader.cs [hunk 1]
```diff
@@ -353,12 +355,14 @@ private bool ParseTag(IByteSource source)
...
-                    if (_stop != null
-                        && _stop(new ParseState { Tag = _tag, SequenceDepth = _sequenceDepth }))
+                    if (_stop != null 
+                        && _stop(new ParseState { PreviousTag = _previousTag, Tag = _tag, SequenceDepth = _sequenceDepth }))
                     {
                         _result = DicomReaderResult.Stopped;
                         return false;
                     }
+                    
+                    _previousTag = _tag;
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/IO/Reader/DicomReader.cs:365]: Unrelated to this PR: I wonder if we can use `_previousTag` to generally check for tags that are less than the previous ones (and ignore these tags). I have previously encountered such data, produced by a big vendor, which crashed dcmtk storescp.

**Review by mrbean-bremen — APPROVED** [review]
Looks good to me.

File: FO-DICOM.Core/IO/Reader/DicomReader.cs (hunk 1 above)
Comment (amoerie) [reply FO-DICOM.Core/IO/Reader/DicomReader.cs:365]: That's exactly the idea that fixed the extraneous file meta information issue that is being solved here :-)

Comment (mrbean-bremen) [reply FO-DICOM.Core/IO/Reader/DicomReader.cs:365]: Yes, but this PR fixes the problem in the meta data, while I meant the same check for the actual dataset (where I have seen that problem).

**Comment by amoerie** [comment]
I suppose this small PR has been open long enough, and there are no objections, so I'll merge this.

**Pull request merged**